remote_mapper.go

```diff
@@ -0,0 +1,196 @@
+package influxdb
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/influxdb/influxdb/influxql"
+)
+
+const (
+	MAX_MAP_RESPONSE_SIZE = 1024 * 1024 * 1024
+)
+
+// RemoteMapper implements the influxql.Mapper interface. The engine uses the remote mapper
+// to pull map results from shards that only exist on other servers in the cluster.
+type RemoteMapper struct {
+	dataNodes []*DataNode
+	resp      *http.Response
+	results   chan interface{}
+	unmarshal influxql.UnmarshalFunc
+	complete  bool
+
+	Call            string   `json:",omitempty"`
```

Comment:
doesn't really matter since these objects are never exported to a user facing API. That convention is for things that go out to users. This is for internal comms only. For example see `ConfigNode` in the raft package.